source/geometry_model/initial_topography_model/prm_polygon.cc

```diff
@@ -0,0 +1,145 @@
+/*
+  Copyright (C) 2011 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/geometry_model/initial_topography_model/prm_polygon.h>
+#include <aspect/utilities.h>
+#include <boost/lexical_cast.hpp>
+
+namespace aspect
+{
+  namespace InitialTopographyModel
+  {
+    template <int dim>
+    double
+    PrmPolygon<dim>::
+    value (const Point<dim-1> &p) const
+    {
+      double value = 0;
+      for (unsigned int i = 0; i < point_lists.size(); i++)
+        {
+          Point<2> p1(p[0],p[1]);
+
+          if (aspect::Utilities::in_2d_polygon<dim>(p1, point_lists[i]))
+            {
+              value = topography_values[i];
+            }
+        }
+      return value;
+    }
+
+
+
+    template <int dim>
+    void
+    PrmPolygon<dim>::
+    declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Initial topography");
+        {
+          prm.enter_subsection("Prm polygon");
+          {
+            prm.declare_entry("Topography parameters",
+                              "",
+                              Patterns::Anything(),
+                              "Set the topography height, end with a |, and set the areas described "
+                              "by the points, separated by commas and coordinates separated by a ':'. "
```

Comment:
This sentence is not correct grammar. Can you also first describe what the parameter does, rather than jumping straight into a description of its format?